component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPUtil.java

```diff
@@ -858,4 +867,116 @@ private static boolean isURLRelative(String contextFromConfig) throws URISyntaxE
+     * @throws TOTPException When handling identity provider configurations.
+     */
+    public static String createDisplayNameForFederatedUsers(AuthenticationContext context, String username)
+            throws TOTPException {
+
+        SequenceConfig sequenceConfig = context.getSequenceConfig();
+        for (Map.Entry<Integer, StepConfig> entry : sequenceConfig.getStepMap().entrySet()) {
+            StepConfig stepConfig = entry.getValue();
+            AuthenticatorConfig authenticatorConfig = stepConfig.getAuthenticatedAutenticator();
+            ApplicationAuthenticator authenticator = authenticatorConfig.getApplicationAuthenticator();
+
+            if (authenticator instanceof FederatedApplicationAuthenticator) {
+                ExternalIdPConfig externalIdPConfig;
+                String externalIdPConfigName = stepConfig.getAuthenticatedIdP();
+                try {
+                    externalIdPConfig = ConfigurationFacade.getInstance()
+                            .getIdPConfigByName(externalIdPConfigName, context.getTenantDomain());
+                } catch (IdentityProviderManagementException e) {
+                    throw new TOTPException(
+                            "TOTP Display name creation failed!. Error while getting External IDP config.", e);
+                }
+                if (stepConfig.isSubjectAttributeStep()) {
+                    if (externalIdPConfig != null) {
+                        Map<String, String> localClaimValues = mapFederateClaimsToLocal(externalIdPConfig, stepConfig, context);
+                        if (localClaimValues.size() == 0 || externalIdPConfig.getIdentityProvider() == null ||
+                                externalIdPConfig.getIdentityProvider().getDefaultAuthenticatorConfig() == null ||
+                                externalIdPConfig.getIdentityProvider().getDefaultAuthenticatorConfig()
+                                        .getDisplayName() == null) {
+                            return null;
+                        }
+
+                        String claimValue;
+                        if (localClaimValues.containsKey(TOTPAuthenticatorConstants.EMAIL_CLAIM_URL)) {
+                            claimValue = localClaimValues.get(TOTPAuthenticatorConstants.EMAIL_CLAIM_URL);
+                        } else if (localClaimValues.containsKey(TOTPAuthenticatorConstants.FIRST_NAME_CLAIM_URL)) {
+                            claimValue = localClaimValues.get(TOTPAuthenticatorConstants.FIRST_NAME_CLAIM_URL);
+                        } else {
+                            claimValue = localClaimValues.getOrDefault(TOTPAuthenticatorConstants.LAST_NAME_CLAIM_URL,
+                                    username);
+                        }
+
+                        String idpType = externalIdPConfig.getIdentityProvider().getDefaultAuthenticatorConfig()
+                                .getDisplayName();
+                        if (claimValue != null && idpType != null) {
```

Comment:
shall we use StringUtils.isNotBlank

Reply:
Address in - https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/109